src/planning.py

```diff
@@ -265,15 +287,18 @@ def _run_low_level_search(task: Task, option_model: _OptionModelBase,
             try:
                 next_state = option_model.get_next_state(state, option)
                 discovered_failures[cur_idx] = None  # no failure occurred
+                if cur_idx >= len(longest_refinement):
+                    longest_refinement = list(plan[:cur_idx + 1])
             except EnvironmentFailure as e:
                 can_continue_on = False
                 failure = _DiscoveredFailure(e, nsrt)
                 # Remember only the most recent failure.
                 discovered_failures[cur_idx] = failure
                 # If we're immediately propagating failures, raise it now.
                 if CFG.sesame_propagate_failures == "immediately":
-                    raise _DiscoveredFailureException("Discovered a failure",
-                                                      failure)
+                    raise _DiscoveredFailureException(
+                        "Discovered a failure", failure,
+                        {"longest_refinement": longest_refinement})
```

Comment:
Confused by the logic here a bit. We'll get into this `except EnvironmentFailure as e:` clause if the call to `option_model.get_next_state(state, option)` above raises that EnvironmentFailure. In that case, `longest_refinement = list(plan[:cur_idx + 1])` won't be run, right?

Reply:
i'm pretty sure this was a logic error, addressed this by making a bunch of changes